pkg/deploy/registry/deploylog/rest.go

```diff
@@ -0,0 +1,237 @@
+	if !ok {
+		return nil, errors.NewBadRequest("namespace parameter required.")
+	}
+	return c.rn.ReplicationControllers(namespace).Watch(lSelector, fSelector, rv)
+}
+
+type podGetter struct {
+	podsNamespacer kclient.PodsNamespacer
+}
+
+// Get is responsible for retrieving the deployer pod
+func (g *podGetter) Get(ctx kapi.Context, name string) (runtime.Object, error) {
+	namespace, ok := kapi.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.NewBadRequest("namespace parameter required.")
+	}
+	return g.podsNamespacer.Pods(namespace).Get(name)
+}
+
+const (
+	defaultTimeout  time.Duration = 10 * time.Second
+	defaultInterval time.Duration = 1 * time.Second
+)
+
+// NewREST creates a new REST for DeploymentLogs. It uses three clients: one for configs,
+// one for deployments (replication controllers) and one for pods to get the necessary
+// attributes to assemble the URL to which the request shall be redirected in order to
+// get the deployment logs.
+func NewREST(
+	dn client.DeploymentConfigsNamespacer,
+	rn kclient.ReplicationControllersNamespacer,
+	pn kclient.PodsNamespacer,
+	connectionInfo kclient.ConnectionInfoGetter) *REST {
+
+	return &REST{
+		Deploy:         &Client{rn: rn, dn: dn},
+		PodGetter:      &podGetter{pn},
+		ConnectionInfo: connectionInfo,
+		Timeout:        defaultTimeout,
+	}
+}
+
+// REST implements GetterWithOptions
+var _ = rest.GetterWithOptions(&REST{})
+
+// NewGetOptions returns a new options object for deployment logs
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &deployapi.DeploymentLogOptions{}, false, ""
+}
+
+// New creates an empty DeploymentLog resource
+func (r *REST) New() runtime.Object {
+	return &deployapi.DeploymentLog{}
+}
+
+// Get returns a streamer resource with the contents of the deployment log
+func (r *REST) Get(ctx kapi.Context, name string, opts runtime.Object) (runtime.Object, error) {
+	deployLogOpts, ok := opts.(*deployapi.DeploymentLogOptions)
+	if !ok {
+		return nil, errors.NewBadRequest("did not get an expected options.")
+	}
+
+	// Fetch deploymentConfig
+	config, err := r.Deploy.GetDeploymentConfig(ctx, name)
+	if err != nil {
+		return nil, errors.NewNotFound("deploymentConfig", name)
+	}
+	// See latest version; if 0, there are no deployments for this config
+	desiredVersion := config.LatestVersion
+	if desiredVersion == 0 {
+		return nil, errors.NewBadRequest(fmt.Sprintf("no deployment exists for deploymentConfig %q", config.Name))
+	}
+	// List deployments
+	deployments, err := r.Deploy.ListDeployments(ctx, deployutil.ConfigSelector(config.Name))
+	if err != nil {
+		return nil, err
+	}
+	if len(deployments.Items) == 0 {
+		return nil, errors.NewInternalError(fmt.Errorf("cannot find deployments for deploymentConfig %q", config.Name))
+	}
+	sort.Sort(deployutil.DeploymentsByLatestVersionDesc(deployments.Items))
+	// Use latest deployment
+	target := &deployments.Items[0]
+	if desiredVersion != deployutil.DeploymentVersionFor(target) {
```

Comment:
I am afraid that this check needs to be duplicated in the client. If we happen to hit it here, how are we going to notify the client via the LocationStreamer that these are not the logs of the deployment they were expecting? @ironcladlou